troubadix/plugins/spaces_before_dots.py

```diff
@@ -0,0 +1,129 @@
+    "common/2009/secpod_novell_edir_mult_vuln_jul09_win.nasl",
+    "common/2010/freebsd/freebsd_3a7c5fc4.nasl",
+    "common/2012/freebsd/freebsd_a4a809d8.nasl",
+    "common/2015/amazon/alas-2014-455.nasl",
+    "common/2015/gb_mozilla_firefox_mult_vuln01_mar15_macosx.nasl",
+    "common/2015/gb_mozilla_firefox_mult_vuln01_mar15_win.nasl",
+    "common/2015/oracle/ELSA-2009-1619.nasl",
+    "common/2015/oracle/ELSA-2011-0586.nasl",
+    "common/2016/gb_perl_privilege_escalation_vuln_win.nasl",
+    "common/2021/dropbear/gb_dropbear_ssh_filename_vuln_may20.nasl",
+    "common/2021/eclipse/gb_jetty_GHSA-v7ff-8wcx-gmc5_lin.nasl",
+    "common/2021/eclipse/gb_jetty_GHSA-v7ff-8wcx-gmc5_win.nasl",
+    "common/gsf/2009/mandriva/gb_mandriva_MDVSA_2008_140.nasl",
+    "common/gsf/2009/mandriva/gb_mandriva_MDVSA_2008_141.nasl",
+    "common/gsf/2010/mandriva/gb_mandriva_MDVA_2010_173.nasl",
+    "common/gsf/2010/mandriva/gb_mandriva_MDVSA_2010_155.nasl",
+    "common/gsf/2010/mandriva/gb_mandriva_MDVSA_2010_155_1.nasl",
+    "common/gsf/2010/mandriva/gb_mandriva_MDVSA_2010_167.nasl",
+    "common/gsf/2020/f5/gb_f5_big_ip_K11315080.nasl",
+    "common/gsf/2020/f5/gb_f5_big_iq_K11315080.nasl",
+    "common/2022/opensuse/gb_opensuse_2022_1548_1.nasl",
+    "common/2024/opensuse/gb_opensuse_2023_3247_1.nasl",
+    "common/attic/debian/deb_232_1.nasl",
+]
+
+
+class CheckSpacesBeforeDots(FileContentPlugin):
+    name = "check_spaces_before_dots"
+
+    def check_content(
+        self, nasl_file: Path, file_content: str
+    ) -> Iterator[LinterResult]:
+        """
+        This plugin checks for excess whitespace before a dot
+        in script_tags that have full sentence values
+        """
+        self.matches = []
+        if nasl_file.suffix == ".inc" or is_ignore_file(nasl_file, IGNORE):
+            return
+        for tag in TAGS:
+            pattern = get_script_tag_pattern(tag)
+            match = pattern.search(file_content)
+            if match:
+                value = match.group("value")
+                value_start = match.start("value")
+
+                for excess_match in PATTERN.finditer(value):
+                    whitespace_pos = excess_match.start() + value_start
+                    self.matches.append((whitespace_pos, excess_match.group()))
+                    fullmatch = match.group()
+                    yield LinterWarning(
+                        f"value of script_tag {match.group('name')} has at least"
+                        " one occurence of excess whitespace before a dot:"
+                        f"\n '{fullmatch}'",
+                        file=nasl_file,
+                        plugin=self.name,
+                    )
+
+    def fix(self) -> Iterator[LinterResult]:
+
+        if not self.matches:
+            return
+
+        # Sort matches by position, descending order to avoid messing up indices during replacement
+        self.matches.sort(reverse=True, key=itemgetter(0))
```

Comment:
```suggestion
        self.matches.sort(reverse=True)
```
`whitespace_pos` is unique, so that there can only ever be one tuple with the same, right?
In that case, we could drop the `key`, because Python will first compare by the first tuple element and only then consider the second (which won't happen, because the first element is unique anyways).
If there are two elements with the same `whitespace_pos`, itemgetter will (probably) keep the original order, instead of then taking the second tuple element into account.